include/libopencm3/stm32/common/lptimer_common_all.h

```diff
@@ -0,0 +1,259 @@
+/** @addtogroup lptimer_defines
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2009 Piotr Esden-Tempski <piotr@esden.net>
+ *
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2009 Piotr Esden-Tempski <piotr@esden.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**@{*/
+
+/** @cond */
+#if defined(LIBOPENCM3_LPTIMER_H)
+/** @endcond */
+#ifndef LIBOPENCM3_LPTIMER_COMMON_H
+#define LIBOPENCM3_LPTIMER_COMMON_H
+
+/* --- LPTIM (low power timer) ------------------------------------------- */
+
+#define LPTIM_ISR(tim_base)		MMIO32((tim_base) + 0x00)
+#define LPTIM_ICR(tim_base)		MMIO32((tim_base) + 0x04)
+#define LPTIM_IER(tim_base)		MMIO32((tim_base) + 0x08)
+#define LPTIM_CFGR(tim_base)		MMIO32((tim_base) + 0x0C)
+#define LPTIM_CR(tim_base)		MMIO32((tim_base) + 0x10)
+#define LPTIM_CMP(tim_base)		MMIO32((tim_base) + 0x14)
+#define LPTIM_ARR(tim_base)		MMIO32((tim_base) + 0x18)
+#define LPTIM_CNT(tim_base)		MMIO32((tim_base) + 0x1C)
+
+#define LPTIM1_ISR			LPTIM_ISR(LPTIM1_BASE)
+#define LPTIM1_ICR			LPTIM_ICR(LPTIM1_BASE)
+#define LPTIM1_IER			LPTIM_IER(LPTIM1_BASE)
+#define LPTIM1_CFGR			LPTIM_CFGR(LPTIM1_BASE)
+#define LPTIM1_CR			LPTIM_CR(LPTIM1_BASE)
+#define LPTIM1_CMP			LPTIM_CMP(LPTIM1_BASE)
+#define LPTIM1_ARR			LPTIM_ARR(LPTIM1_BASE)
+#define LPTIM1_CNT			LPTIM_CNT(LPTIM1_BASE)
+
+#define LPTIM2_ISR			LPTIM_ISR(LPTIM2_BASE)
```

Comment:
pretty sure F410 only has one, shouldn't these either be behind #if defined (LPTIM2_BASE) or moved into the arch specific files?

Reply:
Ah indeed.